docs/architecture/adr-055-protobuf-design.md

```diff
@@ -0,0 +1,45 @@
+# ADR 055: Protobuf Design
+
+## Changelog
+
+- 2020-03-31: Created
+
+## Context
+
+Currently we use [go-amino](https://github.com/tendermint/go-amino) throughout Tendermint. Amino enables quick prototyping and development of features. While this is nice amino does not provide the performance and developer convenience that is expected. For this reason moving to a different encoding format that is widely adopted and supports multiple languages is needed.
```

Comment:
Are the various grievances documented anywhere? I know you've kept at least a mental tally of projects that have run into trouble with Amino, but it could be useful to have the issues thoroughly documented.